// bizcharts/lib/components/Tooltip/ReactTooltip.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _object = _interopRequireDefault(require("object.assign"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _react = _interopRequireDefault(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _get2 = _interopRequireDefault(require("@antv/util/lib/get"));

var _modifyCss2 = _interopRequireDefault(require("@antv/dom-util/lib/modify-css"));

var _withContainer = _interopRequireDefault(require("../../boundary/withContainer"));

var _view = require("../../context/view");

var _core = require("../../core");

var _inner = _interopRequireDefault(require("./inner"));

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var __rest = void 0 && (void 0).__rest || function (s, e) {
  var t = {};

  for (var p in s) {
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
  }

  if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
    if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
  }
  return t;
};

var CONTAINER_CLASS = 'g2-tooltip';

var Tooltip = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2["default"])(Tooltip, _React$Component);

  var _super = _createSuper(Tooltip);

  function Tooltip(props, context) {
    var _this;

    (0, _classCallCheck2["default"])(this, Tooltip);
    _this = _super.call(this, props, context);
    _this.innerContent = /*#__PURE__*/_react["default"].createRef();
    _this.portal = /*#__PURE__*/_react["default"].createRef();
    _this.element = props.container;

    _this.element.classList.add('bizcharts-tooltip');

    _this.element.style.width = 'auto';
    _this.element.style.height = 'auto';
    return _this;
  }

  (0, _createClass2["default"])(Tooltip, [{
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      this.innerContent = null;
    }
  }, {
    key: "overwriteCfg",
    value: function overwriteCfg() {
      var _a = this.props,
          chartView = _a.chartView,
          children = _a.children,
          config = __rest(_a, ["chartView", "children"]);

      var innerContent = this.innerContent;
      chartView.on('tooltip:change', function (_ref) {
        var title = _ref.title,
            items = _ref.items,
            x = _ref.x,
            y = _ref.y;

        if (innerContent.current) {
          innerContent.current.refresh(children(title, items, x, y));
        }

        ;
      });
      chartView.tooltip((0, _object["default"])((0, _object["default"])({
        inPlot: false
      }, config), {
        container: this.element
      }));
      var domStyles = (0, _get2["default"])((0, _core.getTheme)(), ['components', 'tooltip', 'domStyles', CONTAINER_CLASS], {});
      (0, _modifyCss2["default"])(this.element, (0, _object["default"])({}, domStyles));
    }
  }, {
    key: "render",
    value: function render() {
      this.overwriteCfg();
      return /*#__PURE__*/_reactDom["default"].createPortal( /*#__PURE__*/_react["default"].createElement("div", {
        ref: this.portal
      }, /*#__PURE__*/_react["default"].createElement(_inner["default"], {
        ref: this.innerContent
      })), this.element); // 无子组件
    }
  }]);
  return Tooltip;
}(_react["default"].Component);

var _default2 = (0, _withContainer["default"])((0, _view.withView)(Tooltip));

exports["default"] = _default2;
